base/shell/explorer/startmnucust.cpp

```diff
@@ -75,76 +64,112 @@ static VOID OnClearRecentItems(HWND hwnd)
     EnableWindow(GetDlgItem(hwnd, IDC_CLASSICSTART_CLEAR), RecentHasShortcut(hwnd));
 }
 
-struct CUSTOMIZE_ENTRY;
-typedef DWORD (CALLBACK *FN_CUSTOMIZE_READ)(const CUSTOMIZE_ENTRY *entry);
-typedef BOOL (CALLBACK *FN_CUSTOMIZE_WRITE)(const CUSTOMIZE_ENTRY *entry, DWORD dwValue);
+struct CUSTOM_ENTRY;
 
-struct CUSTOMIZE_ENTRY
+typedef BOOL (CALLBACK *FN_CUSTOM_GET)(const CUSTOM_ENTRY *entry);
+typedef VOID (CALLBACK *FN_CUSTOM_SET)(const CUSTOM_ENTRY *entry, BOOL bValue);
+
+struct CUSTOM_ENTRY
 {
     LPARAM id;
     LPCWSTR name;
     BOOL bDefaultValue;
+    FN_CUSTOM_GET fnGetValue;
+    FN_CUSTOM_SET fnSetValue;
     RESTRICTIONS policy1, policy2;
 };
 
-static const CUSTOMIZE_ENTRY s_CustomizeEntries[] =
+static BOOL CALLBACK CustomGetAdvanced(const CUSTOM_ENTRY *entry)
+{
+    return GetAdvancedBool(entry->name, entry->bDefaultValue);
+}
+
+static VOID CALLBACK CustomSetAdvanced(const CUSTOM_ENTRY *entry, BOOL bValue)
+{
+    SetAdvancedDword(entry->name, bValue);
+}
+
+static BOOL CALLBACK CustomGetSmallStartMenu(const CUSTOM_ENTRY *entry)
+{
+    return g_TaskbarSettings.sr.SmallStartMenu;
+}
+
+static VOID CALLBACK CustomSetSmallStartMenu(const CUSTOM_ENTRY *entry, BOOL bValue)
+{
+    g_TaskbarSettings.sr.SmallStartMenu = bValue;
+}
+
+static const CUSTOM_ENTRY s_CustomEntries[] =
 {
     {
         IDS_ADVANCED_DISPLAY_ADMINTOOLS, L"StartMenuAdminTools", TRUE,
+        CustomGetAdvanced, CustomSetAdvanced,
```

Comment:
the final comma could most likely be omitted here and in the other cases, but I see that you just adapted to the existing style. All of our compilers seem to accept that, so just left here as a note. No must to change that.